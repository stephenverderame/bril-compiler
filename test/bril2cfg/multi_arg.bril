@foo(a: int, b: int, c: int) : int {
    x: int = add a b;
    y: int = mul x c;
    ret y;
}

@bar(g: int, f: int) {
    print g;
    print f;
}

@main {
    a: int = const 10;
    b: int = const 20;
    c: int = const 5;
    d: int = call @foo a b c;
    call @bar d d;
}