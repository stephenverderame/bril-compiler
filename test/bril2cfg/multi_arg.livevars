digraph {
	subgraph cluster_foo {
		label="foo(a: int, b: int, c: int)";
		rankdir="TB";
		foo_0 [label="START", shape="rectangle", style="rounded"];
		foo_2 [label="[a, b, c]\n----------\nx := a + b\ny := x * c\nreturn y\n----------\n[]", shape="rectangle", style="rounded"];

		foo_0 -> foo_2;
	}
	subgraph cluster_bar {
		label="bar(g: int, f: int)";
		rankdir="TB";
		bar_0 [label="START", shape="rectangle", style="rounded"];
		bar_2 [label="[f, g]\n----------\nprint g\nprint f\n----------\n[]", shape="rectangle", style="rounded"];

		bar_0 -> bar_2;
	}
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[]\n----------\na := 10\nb := 20\nc := 5\nd := foo(a, b, c)\nbar(d, d)\n----------\n[]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
	}
}
