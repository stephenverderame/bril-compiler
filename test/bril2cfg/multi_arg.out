digraph {
	subgraph cluster_foo {
		label="foo(a: int, b: int, c: int)";
		rankdir="TB";
		foo_0 [label="START", shape="rectangle", style="rounded"];
		foo_2 [label="x := a + b\ny := x * c\nreturn y", shape="rectangle", style="rounded"];

		foo_0 -> foo_2;
	}
	subgraph cluster_bar {
		label="bar(g: int, f: int)";
		rankdir="TB";
		bar_0 [label="START", shape="rectangle", style="rounded"];
		bar_2 [label="print g\nprint f\n", shape="rectangle", style="rounded"];

		bar_0 -> bar_2;
	}
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="a := 10\nb := 20\nc := 5\nd := foo(a, b, c)\nbar(d, d)\n", shape="rectangle", style="rounded"];

		main_0 -> main_2;
	}
}
